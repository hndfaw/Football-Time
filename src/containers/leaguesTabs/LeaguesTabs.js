import React, { Component } from 'react';
import './leaguesTabs.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export class LeaguesTabs extends Component {

  leagueTab = () => {
    const {selectedLeague, leaguesData } = this.props;
    let selectedId = parseInt(selectedLeague.slice(6,9))
    
    const leagueTab = leaguesData.map(league => {
      const btnStyle = (selectedId === league.league_id) ? {background: 'rgba(0,0,0,0.1)'} : null
      return (
        <div className="league-tab" key={league.league_id}>
          <button onClick={this.props.changeOneLeaguesMatches} style={btnStyle} className="league-btn" id={league.league_id}></button>
          <div className="league-tab-inner-container">
            <img src={league.logo} alt="league logo" className="league-tab-logo" />
            <h4 className="league-tab-name">{league.name}</h4>
          </div>
        </div>
      )
    })
    return leagueTab
  }

  render() {
    return  this.leagueTab()
  }
}

LeaguesTabs.propTypes = {
  leaguesData: PropTypes.array,
  selectedLeague: PropTypes.string,
}

export const mapStateToProps = state => ({
  leaguesData: state.leaguesData,
  selectedLeague: state.selectedLeague
})

export default connect(mapStateToProps, null)(LeaguesTabs);
