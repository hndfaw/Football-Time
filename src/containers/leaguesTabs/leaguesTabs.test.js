import React from 'react';
import {LeaguesTabs} from './LeaguesTabs';
import { shallow } from 'enzyme';

describe('LeaguesTabs', () => {
  it('should match the snapshot', () => {
    const mockLeagues = [
      {
        league_id: 524,
        logo: "https://www.api-football.com/public/leagues/2.png",
        name: "Premier League"
      }
    ]

    const wrapper = shallow(<LeaguesTabs selectedLeague="league524" leaguesData={mockLeagues}/>)
    expect(wrapper).toMatchSnapshot()
  })
})