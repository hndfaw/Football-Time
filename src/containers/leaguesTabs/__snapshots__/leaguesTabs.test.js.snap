// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaguesTabs should match the snapshot 1`] = `
<div
  className="league-tab"
  key="524"
>
  <button
    className="league-btn"
    id={524}
    style={
      Object {
        "background": "rgba(0,0,0,0.1)",
      }
    }
  />
  <div
    className="league-tab-inner-container"
  >
    <img
      alt="league logo"
      className="league-tab-logo"
      src="https://www.api-football.com/public/leagues/2.png"
    />
    <h4
      className="league-tab-name"
    >
      Premier League
    </h4>
  </div>
</div>
`;
